shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Genesis/Properties.hs

```diff
@@ -15,18 +27,34 @@ import Test.Tasty.Hedgehog
 prop_golden_ShelleyGenesis :: Property
 prop_golden_ShelleyGenesis = goldenTestJSON exampleShelleyGenesis "test/Golden/ShelleyGenesis"
 
-prop_roundtrip_GenesisDelegationPair_JSON :: Property
-prop_roundtrip_GenesisDelegationPair_JSON =
+prop_roundtrip_Address_JSON :: forall c. Crypto c => Proxy c -> Property
+prop_roundtrip_Address_JSON _ =
+  -- If this fails, FundPair and ShelleyGenesis can also fail.
+  Hedgehog.property $ do
+    addr <- Hedgehog.forAll $ genAddress @c
+    Hedgehog.tripping addr toJSON fromJSON
+    Hedgehog.tripping addr encode decode
+
+prop_roundtrip_GenesisDelegationPair_JSON :: forall c. Crypto c => Proxy c -> Property
+prop_roundtrip_GenesisDelegationPair_JSON _ =
   Hedgehog.property $ do
-    dp <- Hedgehog.forAll genGenesisDelegationPair
+    dp <- Hedgehog.forAll $ genGenesisDelegationPair @c
     Hedgehog.tripping dp toJSON fromJSON
     Hedgehog.tripping dp encode decode
 
-prop_roundtrip_ShelleyGenesis_JSON :: Property
-prop_roundtrip_ShelleyGenesis_JSON = Hedgehog.withTests 500
+prop_roundtrip_FundPair_JSON :: forall c. Crypto c => Proxy c -> Property
```

Comment:
What's a "fund pair"?

Reply:
(Addr c, Coin)

Reply:
OK. Seems weird terminology that's used nowhere else.